src/Calculator/Views/CalculatorProgrammerRadixOperators.xaml

```diff
@@ -490,6 +490,7 @@
                                        AutomationProperties.AutomationId="clearButton"
                                        ButtonId="Clear"
                                        Content="C"
+                                       LostFocus="ClearButton_LostFocus"
                                        Visibility="{x:Bind Model.IsInputEmpty, Mode=OneWay, Converter={StaticResource BooleanToVisibilityConverter}}"/>
 
```

Comment:
Ideally, you want to avoid programmatically handling focus whenever you can. Instead of adding focus management code, can we combine these buttons into one element in XAML that switches between Clear and ClearEntry functionality based on IsInputEmpty?

Reply:
We could do that, it does open up a couple more questions.

Should we combine the two shortcut keys also? ClearEntry has an additional 'delete' shortcut key.
Should we also swap the automation names or combine them into one?

Reply:
> Should we combine the two shortcut keys also?

No, I think we would want to keep separate shortcuts.

> Should we also swap the automation names or combine them into one?

I imagine it would be a single automation name.

> What about in standard mode when they are still two different buttons?

Two separate buttons here, but then you also don't have any funkiness with focus.

> If we want to keep those two things separate I think it is simpler to have two buttons instead of trying to swap out every property.

IMO, swapping out some of the display properties + shortcut is cleaner than wading into focus management. Focus issues are a lot trickier to repro/debug.

Reply:
I agree that is probably a more robust fix. I tried working on your solution by making a general purpose ternary to two generic object converter but hit some issues implementing in C++. I will try again later but I think we should get this fix in soon since it is a big annoyance to customers.